tests/unit/conftest.py

```diff
@@ -117,3 +118,13 @@ def python_cmd_handler(argv: list[str]) -> ExecResult:
 
     yield harness
     harness.cleanup()
+
+
+@pytest.fixture
+def set_environment_variable(request):
+    """Set the environment variables."""
+    environment_variables = request.param
+    os.environ.update(environment_variables)
+    yield
+    for environment_variable in environment_variables:
+        del os.environ[environment_variable]
```

Comment:
Shouldn't you simply keep a copy of the previous env and restore it after yield ?
This will allow overriding existing envrionment variable and restoring them accordingly afterwards instead of removing them

Reply:
I think the [pytest monkeypatch](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.setenv) is designed to handle this and will make sure the environment is restored

Reply:
Updated, but we still need to delete extra environment variables because there isn't a way to replace the entire environment variables.

Reply:
I think monkeypatch has delenv which should restore them after the test

Reply:
Incorporating proxies into the charm state requires introducing a new attribute to the interface of the charm state. However, given its limited utility - being used just once - I think it might not worth it.